common/src/main/java/com/wynntils/wc/custom/item/properties/DurabilityProperty.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wc.custom.item.properties;
+
+import com.wynntils.mc.utils.ItemUtils;
+import com.wynntils.wc.custom.item.WynnItemStack;
+import com.wynntils.wc.utils.WynnUtils;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.item.ItemStack;
+
+public class DurabilityProperty extends ItemProperty {
+    private static final Pattern DURABILITY_PATTERN = Pattern.compile("\\[(\\d+)/(\\d+) Durability\\]");
+
+    private int currentDurability;
+    private int maxDurability;
+
+    public DurabilityProperty(WynnItemStack stack) {
+        super(stack);
+
+        // parse durability
+        List<Component> lore = stack.getOriginalTooltip();
+        for (Component line : lore) {
+            String unformatted = WynnUtils.normalizeBadString(line.getString());
+
+            Matcher durabilityMatcher = DURABILITY_PATTERN.matcher(unformatted);
+            if (!durabilityMatcher.find()) continue;
+
+            this.currentDurability = Integer.parseInt(durabilityMatcher.group(1));
+            this.maxDurability = Integer.parseInt(durabilityMatcher.group(2));
+            break;
+        }
+    }
+
+    public float getDurabilityPct() {
+        return (currentDurability * 1f) / maxDurability;
+    }
+
+    public static boolean hasDurability(ItemStack stack) {
+        for (Component c : ItemUtils.getTooltipLines(stack)) {
+            if (c.getString().endsWith(" Durability]")) return true;
```

Comment:
Why not `ItemUtils.getStringLore()` then do a `.contains()`?

Reply:
ah, i forgot that we had that

looking over the ItemUtils stuff i'm not a big fan of how it implements all those lore functions though - directly getting the NBT data really isn't necessary

Reply:
Fair, though I still think this check should be improved - wouldn't the `.endsWith()` trip when a user has a custom lore on their item? Or is the `]` invalid for a custom lore?

Reply:
The more precise the requirements, the better I think the check is. If you were to just look of the lore contained "Durability" anywhere at all, it would fail at the "Dwarfen Durability Enhancing Ring" or whatever. So removing the trailing `]` is not an improvement, quite the opposite.